test/DecryptionFailureTracker-test.ts

```diff
@@ -274,24 +279,22 @@ describe('DecryptionFailureTracker', function() {
 
         tracker.trackFailures();
 
-        //expect(counts['UnknownError']).toBe(1, 'should track one UnknownError');
-        expect(counts['OlmKeysNotSentError']).toBe(2, 'should track two OlmKeysNotSentError');
+        // called once for each error type
+        expect(trackEventSpy).toHaveBeenCalledTimes(2);
+        expect(trackEventSpy).toHaveBeenCalledWith('E2E', 'Decryption failure', 'UnknownError', '1');
+        expect(trackEventSpy).toHaveBeenCalledWith('E2E', 'Decryption failure', 'OlmKeysNotSentError', '2');
     });
 
     it('should aggregate error codes correctly', () => {
-        const counts = {};
-        const tracker = new DecryptionFailureTracker(
-            (total, errorCode) => counts[errorCode] = (counts[errorCode] || 0) + total,
-            (errorCode) => 'OlmUnspecifiedError',
-        );
+        const tracker = DecryptionFailureTracker.instance;
 
         const decryptedEvent1 = createFailedDecryptionEvent();
         const decryptedEvent2 = createFailedDecryptionEvent();
         const decryptedEvent3 = createFailedDecryptionEvent();
 
-        const error1 = new MockDecryptionError('ERROR_CODE_1');
-        const error2 = new MockDecryptionError('ERROR_CODE_2');
-        const error3 = new MockDecryptionError('ERROR_CODE_3');
+        const error1 = new MockDecryptionError(undefined);
+        const error2 = new MockDecryptionError(undefined);
+        const error3 = new MockDecryptionError(undefined);
```

Comment:
I think this was testing the unknown error code behaviour? `undefined` does aggregate, but so should the 3 stringy error codes (according to the test)

Reply:
@duxovni is this the correct behaviour? Should `DecryptionFailureTracker` actually being aggregating the errors my their mapped code before calling the tracking function in `trackFailures`? Feels like the description of the test and the actual behaviour are at odds

Reply:
Reverted in favour of #8272